editors/vscode/server/src/server.ts

```diff
@@ -283,24 +285,21 @@ documents.onDidChangeContent((
 )());
 
 
-function convertSpan(index: number, text: string): Position {
-	const buffer = new TextEncoder().encode(text);
-	let line = 0;
-	let character = 0;
-
-	let i = 0;
-	while (i < text.length) {
-		if (i == index) {
-			return { line, character };
-		}
-
-		if (buffer.at(i) == 0x0A) {
-			line++;
-			character = 0;
-		} else {
-			character++;
+function convertSpan(utf8_offset: number, utf16_text: string, lineBreaks: Array<number>): Position {
+	let line = undefined;
+	for (let n = 0; n < lineBreaks.length; ++n) {
```

Comment:
Here instead of a linear search we could do a binary one since line breaks are in order.

Reply:
We could absolutely do that :)
I'll leave that for our future selves to do.